platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ImportMigrationConfigurationAdminContext.java

```diff
@@ -0,0 +1,234 @@
+                        .stream(),
+                exportedFactoryServices.values()
+                        .stream()
+                        .flatMap(List::stream));
+    }
+
+    @Override
+    public ImportMigrationEntry getEntry(Path path) {
+        throw new UnsupportedOperationException("should not be called");
+    }
+
+    @Override
+    public Stream<ImportMigrationEntry> entries(Path path) {
+        throw new UnsupportedOperationException("should not be called");
+    }
+
+    @Override
+    public Stream<ImportMigrationEntry> entries(Path path, PathMatcher filter) {
+        throw new UnsupportedOperationException("should not be called");
+    }
+
+    // PMD.DefaultPackage - designed to be called only from ImportMigrationConfigurationAdminEntry within this package
+    @SuppressWarnings("PMD.DefaultPackage")
+    List<Configuration> getMemoryFactoryService(String factoryPid) {
+        return memoryFactoryServices.get(factoryPid);
+    }
+
+    // PMD.DefaultPackage - designed to be called only from ImportMigrationConfigurationAdminEntry within this package
+    @SuppressWarnings("PMD.DefaultPackage")
+    void removeMemoryFactoryService(String factoryPid) {
+        memoryFactoryServices.remove(factoryPid);
+    }
+
+    // PMD.DefaultPackage - designed to be called only from ImportMigrationConfigurationAdminEntry within this package
+    @SuppressWarnings("PMD.DefaultPackage")
+    Configuration removeMemoryService(String pid) {
+        return memoryServices.remove(pid);
+    }
+
+    // PMD.DefaultPackage - designed to be called only from ImportMigrationConfigurationAdminEntry within this package
+    @SuppressWarnings("PMD.DefaultPackage")
+    Configuration createConfiguration(ImportMigrationConfigurationAdminEntry entry)
+            throws IOException {
+        // Question: should we use the bundle location that was exported???
+        // If we do, should we perform additional checks to make sure we're not loading a malicious bundle?
+        // This might be unnecessary if we are comfortable with the encryption of the zip file as our only countermeasure.
+        if (entry.isManagedServiceFactory()) {
+            return configurationAdmin.createFactoryConfiguration(entry.getFactoryPid(), null);
+        }
+        return configurationAdmin.getConfiguration(entry.getPid());
+    }
+
+    private ImportMigrationConfigurationAdminEntry proxy(ImportMigrationEntry entry) {
+        final Path path = entry.getPath();
+        final String extn = FilenameUtils.getExtension(path.toString());
+        final PersistenceStrategy ps = admin.getPersister(extn);
+
+        if (ps == null) {
+            getReport().record(new MigrationException(
+                    "Import error: persistence strategy [%s] for configuration [%s] is not defined.",
+                    extn,
+                    path));
+        } else {
+            final Dictionary<String, Object> properties;
+            InputStream is = null;
```

Comment:
😄 Because I do not care to get exceptions out of a close for an input stream and try-resource does throw that exception out if it fails to close. On an output stream, the story would be totally different and I would care about these exceptions.